### PR TITLE
EVG-5965: add logs to debug where cached task is not updated

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-5965

### CODE REVIEW

**Comment by john-m-liu** [comment]
I think this is a fine change to make but I don't think it'll help. The spots that you're logging return errors, which almost certainly would already have been logged elsewhere if they were hit. If you searched for the error strings in splunk and didn't find them, what I would do is trace a couple of specific tasks through what should have updated the cache, and add logging in the area that should have updated it

File: model/task_lifecycle.go [hunk 1]
```diff
@@ -576,6 +592,14 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
...
 		err = b.SetCachedTaskFinished(t.Id, status, &t.Details, t.TimeTaken)
 		if err != nil {
+			grip.Error(message.Fields{
+				"message":    "failed to set cached task finished",
+				"function":   "UpdateBuildAndVersionStatusForTask",
+				"build_id":   b.Id,
+				"task_id":    t.Id,
+				"status":     status,
+				"time_taken": t.TimeTaken,
+			})
```
Comment (tychoish) [inline model/task_lifecycle.go]: I'd make all of these calls be message.WrapErrors(err, message.Fields{})

File: model/task_lifecycle.go [hunk 2]
```diff
@@ -576,6 +592,14 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
...
 			return fmt.Errorf("error updating build: %v", err.Error())
```
Comment (tychoish) [inline model/task_lifecycle.go]: this really should be an errors.Wrap(err, "") situation

File: model/task_lifecycle.go [hunk 3]
```diff
@@ -423,11 +423,27 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
...
 		if err = build.UpdateCachedTask(t.DisplayTask.BuildId, t.DisplayTask.Id, t.DisplayTask.Status, t.TimeTaken); err != nil {
+			grip.Error(message.Fields{
+				"message":    "failed to update cached display task",
+				"function":   "MarkEnd",
+				"build_id":   t.DisplayTask.BuildId,
+				"task_id":    t.DisplayTask.Id,
+				"status":     t.DisplayTask.Status,
+				"time_taken": t.TimeTaken,
+			})
 			return err
```
Comment (tychoish) [inline model/task_lifecycle.go]: wrap this error too?

Comment (Kimchelly) [reply model/task_lifecycle.go]: Done.

File: model/task_lifecycle.go (hunk 1 above)
Comment (Kimchelly) [reply model/task_lifecycle.go]: Done.

File: model/task_lifecycle.go (hunk 2 above)
Comment (Kimchelly) [reply model/task_lifecycle.go]: Done.

**Comment by tychoish** [comment]
I think wrapping the errors will give you a better stack trace

**Pull request merged**